File: julia/primes/pnt.jl
using Primes, Plots, LinearAlgebra

P=zeros(100000);
P[1]=1;
P[2]=1;
L=zeros(100000);
L[1]=3/log(3);
L[2]=3/log(3);
N=zeros(100000);
N[1]=1;
N[2]=2;
Q=ones(100000);

for i=3:100000
    A=zeros(i);
    for j=1:i
        A[j]=Float64(isprime(j));
    end
    P[i]=sum(A);
    L[i]=i/log(i);
    N[i]=i;
end

p1=plot([P L],  
    grid=false,
    linewidth=2, 
    title="Prime Number Theorem: Pi(x) and x/log(x)", 
    right_margin=Plots.Measures.Length(:mm, 10.0),
    left_margin=Plots.Measures.Length(:mm, 5.0),
    xlabel="x",
    xticks = ([1 50000 100000;], ["e" "50000" "100000"]),
    yaxis="Number of Primes",
    legendfont=font(10), 
    label=["Pi(x)" "x/log(x)"],
    palette = :seaborn_bright, 
    legend = :topleft)
savefig("pnt1.png") 

p2=plot([P./L Q],  
    grid=false,
    linewidth=2, 
    title="Prime Number Theorem: Pi(x)/(x/log(x))", 
    right_margin=Plots.Measures.Length(:mm, 10.0),
    left_margin=Plots.Measures.Length(:mm, 5.0),
    xlabel="x",
    xticks = ([1 50000 100000;], ["e" "50000" "100000"]),
    yaxis="",
    yticks = ([0.4 0.8 1.0 1.2;], ["0.4" "0.8" "1.0" "1.2"]),
    label = :false,
    palette = :seaborn_bright, 
    legend = :topright)
savefig("pnt2.png") 

plot(p1, p2,
     layout=(1,2), 
     size=(1200,400), 
     left_margin=Plots.Measures.Length(:mm, 5.0),
     right_margin=Plots.Measures.Length(:mm, 15.0),
     top_margin=Plots.Measures.Length(:mm, 5.0),
     bottom_margin=Plots.Measures.Length(:mm, 5.0))
savefig("pnt.png") 

File: MATLAB/primes/pnt.m
P=zeros(1,10000);
P(1,1)=1;
P(1,2)=1;
L=zeros(1,10000);
L(1,1)=3/log(3);
L(1,2)=3/log(3);
N=zeros(1,10000);
N(1,1)=1;
N(1,2)=2;
Q=zeros(1,10000);
Q=Q+1;

for i=3:10000
    A=zeros(1,i);
    for j=1:i
        A(1,j)=double(isprime(j));
    end
    P(1,i)=sum(A(1,:));
    L(1,i)=i/log(i);
    N(1,i)=i;
end

newcolors = [0 0 1; 
             0 1 0; 
             1 0 0];
colororder(newcolors)

subplot(1,2,1)
plot(N,P,N,L,'LineWidth',2)
title('The Prime Number Theorem: Pi(x) and x/log(x)')
xlabel('x');
xticks([1 5000 10000]);
xticklabels({'e','5000','10000'});
legend('Pi(x)','x/log(x)','Location','northwest');

subplot(1,2,2)
plot(N,P./L,N,Q,'LineWidth',2)
title('The Prime Number Theorem: Pi(x)/(x/log(x))')
xlabel('x');
xticks([1 5000 10000]);
xticklabels({'e','5000','10000'});

set(gcf,'Position',[600,200,1200,400]);
saveas(gcf,'pnt.png');